# Random Forest Regressor - stepwise best model

## Import Packages

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline
from sklearn.ensemble import RandomForestRegressor
from sklearn.model_selection import train_test_split
from sklearn.metrics import mean_absolute_error
from sklearn.model_selection import cross_val_score, GridSearchCV, cross_val_predict
from sklearn.preprocessing import MinMaxScaler

## Import Data

In [2]:
coffee3 = pd.read_csv("./coffee3.csv")

In [3]:
list(coffee3)

['Country.of.Origin',
 'Region.Name',
 'Processing.Method',
 'Aroma',
 'Flavor',
 'Aftertaste',
 'Acidity',
 'Body',
 'Balance',
 'Uniformity',
 'Clean.Cup',
 'Sweetness',
 'Cupper.Points',
 'Total.Cup.Points',
 'altitude_mean_meters']

## Data Wrangling - subset X variables & Y variables in own arrays
### x variable based on best stepwise model ran in R

In [4]:
x = coffee3[['Flavor', 'Aftertaste', 'Acidity', 'Body', 'Balance', 'Clean.Cup']]
y = coffee3[['Cupper.Points']]

## Train Test Split

In [5]:
x_train, x_test, y_train, y_test = train_test_split(x,y, test_size=0.3, random_state=13)

## Create Random Forest Model

In [6]:
forest = RandomForestRegressor(n_estimators=500, random_state=13)

In [7]:
forest.fit(x_train, y_train)

C:\Users\timca\AppData\Local\Temp\ipykernel_3008\2365454903.py:1: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  forest.fit(x_train, y_train)


RandomForestRegressor(n_estimators=500, random_state=13)

In [8]:
print("Score:", forest.score(x_test, y_test))

Score: 0.6444862069446246


In [9]:
scores = cross_val_score(forest, x, y, cv=10, scoring='neg_mean_absolute_error')

C:\Python3\lib\site-packages\sklearn\model_selection\_validation.py:686: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  estimator.fit(X_train, y_train, **fit_params)
C:\Python3\lib\site-packages\sklearn\model_selection\_validation.py:686: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  estimator.fit(X_train, y_train, **fit_params)
C:\Python3\lib\site-packages\sklearn\model_selection\_validation.py:686: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the shape of y to (n_samples,), for example using ravel().
  estimator.fit(X_train, y_train, **fit_params)
C:\Python3\lib\site-packages\sklearn\model_selection\_validation.py:686: DataConversionWarning: A column-vector y was passed when a 1d array was expected. Please change the

In [10]:
print(scores)

[-0.32524179 -0.21730481 -0.13509369 -0.1377109  -0.11757445 -0.14978545
 -0.20074099 -0.18751428 -0.14592994 -0.26357117]


In [11]:
np.mean(scores)

-0.1880467476988934

## Random Forest Regressor can predict Cupper.Point witin .19